application/src/main/webapp/js/chat-modules.js

```diff
@@ -389,11 +393,36 @@ ChatRoom.prototype.getUserLastMessage = function() {
 /**
  * Show Messages (json to html)
  * @param msgs : json messages data to show
+ * @param merge : if true merge msgs with the current list, else replace
  */
-ChatRoom.prototype.showMessages = function(msgs) {
+ChatRoom.prototype.showMessages = function(msgs, merge) {
```

Comment:
ok it works if we add "this.messages=[];" in the init method. So lastTS ==0  is equivalent to this.message is empty 
i've just realized that ChatRoom is a singleton, not an instance for each room, so init method have to reset all variables.